### PR TITLE
Issue #497 Finalize Serializer documentation and samples

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

File: impl/src/test/java/org/ehcache/docs/GettingStarted.java [hunk 1]
```diff
@@ -339,7 +342,7 @@ public void cacheSerializingCopiers() throws Exception {
     CacheConfiguration<Long, Person> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder()
         .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder().heap(10, EntryUnit.ENTRIES).build())
         .add(new DefaultCopierConfiguration<Long>((Class)SerializingCopier.class,   //<1>
```
Comment (albinsuresh) [inline impl/src/test/java/org/ehcache/docs/GettingStarted.java]: `DefaultCopierConfiguration<Person>` right??? Since you changed the copier config type.

Comment (ljacomet) [inline impl/src/test/java/org/ehcache/docs/GettingStarted.java]: Good catch - fixing that!

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 2]
```diff
@@ -217,15 +223,29 @@ include::{sourcedir}/impl/src/test/java/org/ehcache/docs/GettingStarted.java[tag
...
 <2> Configure the value's serializer.
 
-=== Store-By-Value Caches with Copiers
+===== Implementation of a `Serializer`
+
+When implementing a `org.ehcache.spi.serialization.Serializer`, you obviously need to respect the interface contract.
+In addition, there are a few implicit contracts:
+
+* Your implementation must have a public constructor taking a `ClassLoader` as parameter.
+* There are two annotations that you have to use to mark your `Serializer` - a single implementation can use both:
```
Comment (albinsuresh) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: "have to use" or "may use"? The annotations are not mandatory when configured at the cache level, right? It is mandatory for the cache manager level config, though.

Comment (ljacomet) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Well, _have to_ covers both :smiley:

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 3]
```diff
@@ -185,15 +188,18 @@ include::{sourcedir}/impl/src/test/java/org/ehcache/docs/GettingStarted.java[tag
...
-multi-tiered cache, serialization and deserialization happens when the data is moved between the tiers. Ehcache provides
-a serializer that can serialize any `Serializable` type. So if you are using standard `Serializable` types as your keys
-and values, no extra serializers need to be configured as the inbuilt serializer would handle those cases. But if you
-use your own custom types in ehcache that are not serializable, the stock serializer won't be able to perform the
-serialization/deserialization on its own. So you need to provide the serialization logic as well. Inorder to do this you
-can register your own custom serializers to the cache/cache manager. So if you register a serializer for some type of
-data, that serializer would be used to perform all the serialization/deserialization for that type.
-You can configure a set of serializers on the cache manager:
+Serialization is primarily used in ehcache to store the data in the non-heap tiers.
+So whenever the user uses a multi-tiered cache, serialization and deserialization happens when the data is moved to, from and between the tiers.
```
Comment (AbfrmBlr) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: This shouldn't be always true after #369 is done

Comment (ljacomet) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Maybe, but it remains an implementation detail while indicating that here hints that you want a performant serialisation solution.

Comment (lorban) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: True, I'd reword that as '… serialization and deserialization *can happen* when …'
